src/c++/perf_analyzer/test_command_line_parser.cc

```diff
@@ -0,0 +1,1040 @@
+  {
+    int argc = 5;
+    char* argv[argc] = {app_name, "-m", model_name, "--concurrency-range",
+                        "100,400,10"};
+
+    REQUIRE_NOTHROW(act = parser.Parse(argc, argv));
+    CHECK(!parser.UsageCalled());
+
+    // BUG: Should detect this and through an error. User will enter this and
+    // have no clue why the end and step sizes are not used correctly.
+    //
+
+    exp->using_concurrency_range = true;
+    exp->concurrency_range.start = 100;
+    CHECK_PARAMS(act, exp);
+    optind = 1;
+  }
+
+  SUBCASE("bad start value")
+  {
+    int argc = 5;
+    char* argv[argc] = {app_name, "-m", model_name, "--concurrency-range",
+                        "bad:400:10"};
+
+    REQUIRE_NOTHROW(act = parser.Parse(argc, argv));
+    CHECK(parser.UsageCalled());
+    CHECK_STRING(
+        "Usage Message", parser.GetUsageMessage(),
+        "failed to parse concurrency range: bad:400:10");
+
+    exp->using_concurrency_range = true;
+    CHECK_PARAMS(act, exp);
+    optind = 1;
+  }
+
+  SUBCASE("bad end value")
+  {
+    int argc = 5;
+    char* argv[argc] = {app_name, "-m", model_name, "--concurrency-range",
+                        "100:bad:10"};
+
+    REQUIRE_NOTHROW(act = parser.Parse(argc, argv));
+    CHECK(parser.UsageCalled());
+    CHECK_STRING(
+        "Usage Message", parser.GetUsageMessage(),
+        "failed to parse concurrency range: 100:bad:10");
+
+    exp->using_concurrency_range = true;
+    exp->concurrency_range.start = 100;
+    CHECK_PARAMS(act, exp);
+    optind = 1;
+  }
+
+  SUBCASE("bad step value")
+  {
+    int argc = 5;
+    char* argv[argc] = {app_name, "-m", model_name, "--concurrency-range",
+                        "100:400:bad"};
+
+    REQUIRE_NOTHROW(act = parser.Parse(argc, argv));
+    CHECK(parser.UsageCalled());
+    CHECK_STRING(
+        "Usage Message", parser.GetUsageMessage(),
+        "failed to parse concurrency range: 100:400:bad");
+
+    exp->using_concurrency_range = true;
+    exp->concurrency_range.start = 100;
+    exp->concurrency_range.end = 400;
+    CHECK_PARAMS(act, exp);
+    optind = 1;
+  }
+
+  SUBCASE("invalid condition - end and latency threshold are 0")
+  {
+    int argc = 7;
+    char* argv[argc] = {app_name,   "-m",
+                        model_name, "--concurrency-range",
+                        "100:0:25", "--latency-threshold",
+                        "0"};
+
+    REQUIRE_NOTHROW(act = parser.Parse(argc, argv));
+    CHECK(parser.UsageCalled());
+    CHECK_STRING(
+        "Usage Message", parser.GetUsageMessage(),
+        "The end of the search range and the latency limit can not be both 0 "
+        "(or 0.0) simultaneously");
+
+    exp->using_concurrency_range = true;
+    exp->concurrency_range.start = 100;
+    exp->concurrency_range.end = 0;
+    exp->concurrency_range.step = 25;
+    exp->latency_threshold_ms = 0;
+    CHECK_PARAMS(act, exp);
+    optind = 1;
+  }
+
+  // TODO figure out how to setup an asynchronous sequence model
```

Comment:
Can this todo be turned into a ticket and the starter code be put in that ticket?

Reply:
> When I am done with the test_command_line_parser.cc

I'm having difficulty understanding. Are you intending to add commented/starter code into the `main` branch?

Reply:
I removed the commented section for this PR.